### PR TITLE
Add support for UTF8 Feature Extension

### PR DESCRIPTION
**Pull request by keeratsingh**
This PR adds support for `UTF-8 Feature Extension` that will be introduced in newer versions of SQL Server.

### CODE REVIEW

File: src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs [hunk 1]
```diff
@@ -2282,13 +2282,23 @@ private bool TryProcessEnvChange(int tokenLength, TdsParserStateObject stateObj,
...
-                            int newCodePage = GetCodePage(env.newCollation, stateObj);
-                            if (newCodePage != _defaultCodePage)
+                            // UTF8 collation
+                            if ((env.newCollation.info & TdsEnums.UTF8_IN_TDSCOLLATION) == TdsEnums.UTF8_IN_TDSCOLLATION)
                             {
-                                _defaultCodePage = newCodePage;
-                                _defaultEncoding = System.Text.Encoding.GetEncoding(_defaultCodePage);
+                                _defaultEncoding = Encoding.UTF8;
+                                _defaultCollation = env.newCollation;
+                                _defaultLCID = env.newCollation.LCID;
```
Comment (udlingal) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs]: this is redundant here

**Comment by stephentoub** [comment]
cc: @GrabYourPitchforks

File: src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs (hunk 1 above)
Comment (keeratsingh) [reply src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs]: Removed line.

**Comment by keeratsingh** [comment]
@dotnet-bot Test Windows x64 Debug Build

File: src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs [hunk 2]
```diff
@@ -2282,13 +2282,22 @@ private bool TryProcessEnvChange(int tokenLength, TdsParserStateObject stateObj,
...
-                            _defaultCollation = env.newCollation;
-                            int newCodePage = GetCodePage(env.newCollation, stateObj);
-                            if (newCodePage != _defaultCodePage)
+                            // UTF8 collation
+                            if ((env.newCollation.info & TdsEnums.UTF8_IN_TDSCOLLATION) == TdsEnums.UTF8_IN_TDSCOLLATION)
                             {
-                                _defaultCodePage = newCodePage;
-                                _defaultEncoding = System.Text.Encoding.GetEncoding(_defaultCodePage);
+                                _defaultEncoding = Encoding.UTF8;
+                                _defaultCollation = env.newCollation;
```
Comment (saurabh500) [inline src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs]: `_defaultCollation = env.newCollation`
This is set in both if and else. Can we move it out to before if and else.

Comment (keeratsingh) [reply src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs]: Moved it out before if and else as requested.

**Comment by keeratsingh** [comment]
@dotnet-bot Test Windows x64 Debug Build

**Pull request merged**